flang/lib/Lower/IntrinsicCall.cpp

```diff
@@ -1962,6 +1967,46 @@ void IntrinsicLibrary::genCpuTime(llvm::ArrayRef<fir::ExtendedValue> args) {
   builder.create<fir::StoreOp>(loc, res2, *arg);
 }
 
+// CSHIFT
+fir::ExtendedValue
+IntrinsicLibrary::genCshift(mlir::Type resultType,
+                            llvm::ArrayRef<fir::ExtendedValue> args) {
+  assert(args.size() == 3);
+
+  // Handle required ARRAY argument
+  auto array = builder.createBox(loc, args[0]);
+  fir::BoxValue arrayTmp = builder.createBox(loc, args[0]);
```

Comment:
Prefer the pattern:
```
fir::BoxValue arrayBox= builder.createBox(loc, args[0]);
auto array = fir::getBase(arrayBox);
```

`createBox` generates IR each time it is called. The snippet above avoids generating the same IR twice just to get the rank. Most of this duplicated IR is pruned by dead code elimination before you see it, but here it is really easy to not do extra work in the first place.

Reply:
Thanks, Jean.  I'll fix it.